src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -290,9 +310,13 @@ def _process_resource_change(
 def process_resource_change_bound(
     self: Task, action: str, sender: str, instance_id: int, **kwargs: Any
 ) -> None:
-    _process_resource_change(
-        action=action, sender=sender, instance_id=instance_id, retryer=self, **kwargs
-    )
+    try:
+        _process_resource_change(
+            action=action, sender=sender, instance_id=instance_id, retryer=self, **kwargs
+        )
+    except (AssertionError, SentryAppSentryError) as e:
+        sentry_sdk.capture_exception(e)
+        return
```

Comment:
i don't think these need to be explicitly captured, iirc when they are raised in the child function, since they aren't caught there they should be bubbled up -> sent to Sentry, and the `retry_decorator` doesn't retry on these errors anyway

Reply:
retry decorator doesn't capture ignored exceptions so if we want a Sentry error we need to capture in func

Reply:
sadge https://github.com/getsentry/sentry/blob/1a623d6997a66c10ec8ad66fee9a77372c54670e/src/sentry/tasks/base.py#L157-L170